### PR TITLE
Fix render of tile #62, fix town revenue on pointy

### PR DESCRIPTION
**Pull request by jenf**
Fixes rendering of tile #62 to
Screenshot from 2020-06-17 17-24-18

This also improves the rendering of multi-city multi-slot tiles generally

Town revenues are now not rotated by 30 degrees

### CODE REVIEW

File: assets/app/view/game/part/city.rb [hunk 1]
```diff
@@ -85,24 +94,25 @@ class City < Base
...
+          [[5, 12], false],
+          [[4], true],
+          [[11, 18], true],
+          [[11, 18], false],
         ].freeze
 
         def preferred_render_locations
           if @num_cities > 1 && @edge
+            weights = EDGE_TRACK_REGIONS[@edge] + EDGE_CITY_REGIONS[@edge]
+            weights += EXTRA_SLOT_REGIONS[@edge] if @city.slots != 1
```
Comment (tobymao) [inline assets/app/view/game/part/city.rb:109]: unless @city.slots.one?

Comment (tobymao) [reply assets/app/view/game/part/city.rb:109]: weights.concat perhaps?

File: assets/app/view/game/part/city.rb [hunk 2]
```diff
@@ -138,13 +147,16 @@ def load_from_tile
...
+            slot_rotation = (360 / @city.slots) * slot_index
 
             # use the rotation on the outer <g> to position the slot, then use
             # -rotation on the Slot so its contents are rendered without
             # rotation
-            h(:g, { attrs: { transform: "rotate(#{rotation})" } }, [
-              h(:g, { attrs: { transform: "#{translate} rotate(#{-rotation})" } }, [
+            x, y = CITY_SLOT_POSITION[@city.slots]
+            revert_angle = render_location[:angle]
+            revert_angle += angle_for_layout unless @num_cities > 1 && @edge
```
Comment (tobymao) [inline assets/app/view/game/part/city.rb:157]: never use unless with multiple statements

File: assets/app/view/game/part/city.rb [hunk 3]
```diff
@@ -205,20 +209,19 @@ def render_revenue
                       end
           when 2
             if @edge
-              angle = OO_REVENUE_ANGLES[@edge]
-              regions = OO_REVENUE_REGIONS[@edge]
+              regions, negative_displacement = OO_REVENUE_REGIONS[@edge]
+              displacement = - displacement if negative_displacement
```
Comment (tobymao) [inline assets/app/view/game/part/city.rb:213]: displacement *= -1 if negative

Comment (tobymao) [reply assets/app/view/game/part/city.rb:213]: or displacement = -displacement

File: assets/app/view/game/part/label.rb [hunk 4]
```diff
@@ -16,25 +16,25 @@ class Label < Base
...
+            region_weights: { LEFT_CORNER => 1.0, LEFT_MID => 0.25 },
             x: -67,
             y: 0,
           },
         }.freeze
 
         P_BOTTOM_LEFT_CORNER = {
           flat: {
-            region_weights: { [19, 20] => 1.0, [21] => 0.5 },
+            region_weights: { BOTTOM_LEFT_CORNER => 1.0, [21] => 0.5 },
```
Comment (tobymao) [inline assets/app/view/game/part/label.rb:32]: i think you're going to conflict with me here

File: assets/app/view/game/tile.rb [hunk 5]
```diff
@@ -46,7 +46,8 @@ def should_render_revenue?
 
         return false if revenue.uniq.size > 1
 
-        return false if revenue.size == 2
+        total_slots = @tile.cities.sum(&:slots)
+        return false if total_slots < 3 && revenue.size == 2
```
Comment (tobymao) [inline assets/app/view/game/tile.rb:50]: can just make this a one liner

File: assets/app/view/game/part/city.rb [hunk 6]
```diff
@@ -16,6 +16,7 @@ class City < Base
 
         needs :tile
         needs :city
+        needs :should_render_revenue
```
Comment (tobymao) [inline assets/app/view/game/part/city.rb:19]: so why is this needed now and the old method not working?

Comment (jenf) [reply assets/app/view/game/part/city.rb:19]: This is the first tile we've had with multicity multislot, I didn't want to dup the logic in two places

Comment (tobymao) [reply assets/app/view/game/part/city.rb:19]: gotcha, did you clean up all the duplicate logic?

Comment (tobymao) [reply assets/app/view/game/part/city.rb:19]: can you rename it render_revenue or show_revenue ?

Comment (jenf) [reply assets/app/view/game/part/city.rb:19]: Yep, I cleared most of it. There's a little safety code I left in. Will rename to show revenue

File: assets/app/view/game/part/city.rb (hunk 1 above)
Comment (jenf) [reply assets/app/view/game/part/city.rb:109]: one? is enumerable only

File: assets/app/view/game/part/label.rb (hunk 4 above)
Comment (jenf) [reply assets/app/view/game/part/label.rb:32]: Merged with master, no conflicts. I think I merged with your changes
